### PR TITLE
use newer `listObjectsV2` method and add prop types

### PR DESCRIPTION
**Pull request by cabhishek**
* Use `listObjectsV2` method from `aws-sdk`
* Add `home` link to breadcrum nav bar
* Empty paths are handled as root dir path 
* Prop types for sanity

### CODE REVIEW

File: packages/commuter-breadcrumb/src/BreadCrumb.js [hunk 1]
```diff
@@ -1,37 +1,43 @@
...
+  breadCrumbs.push(
+    <Breadcrumb.Section key="home">
+      <Link to="/">
+        Home
+      </Link>
+    </Breadcrumb.Section>
+  );
   paths.forEach((name, index) => {
     const fullPath = paths.slice(0, index + 1).join("/");
+    breadCrumbs.push(
```
Comment (jdetle) [inline packages/commuter-breadcrumb/src/BreadCrumb.js]: Why the name breadcrumbs? Is that a term I'm unfamiliar with?

Comment (cabhishek) [inline packages/commuter-breadcrumb/src/BreadCrumb.js]: @jdetle  http://react.semantic-ui.com/collections/breadcrumb

File: package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "client": "lerna exec --scope commuter-client -- npm run start",
     "clean:install": "lerna clean",
     "clean:build": "lerna exec -- rm -rf ./build",
-    "format:code": "prettier --write ./packages/*/src/**/*.js",
+    "format:code": "prettier --write ./packages/*/src/*/*.js ./packages/*/src/*.js",
```
Comment (rgbkrk) [inline package.json]: You can probably swap this with `prettier --write ./packages/*/src/**/*.js` (note the `**`).

Comment (cabhishek) [inline package.json]: yeah I had `prettier --write ./packages/*/src/**/*.js` previously. But subfolders were not getting formatted recursively. Also, `lerna exec -- prettier --write ./src/**/*.js` fails.

Comment (rgbkrk) [inline package.json]: :man_facepalming: My apologies, I see that now.

**Pull request merged**